core/src/test/java/org/elasticsearch/search/child/ChildQuerySearchIT.java

```diff
@@ -1926,4 +1927,51 @@ public void testHasChildInnerQueryType() {
                 QueryBuilders.hasChildQuery("child-type", new IdsQueryBuilder().addIds("child-id"))).get();
         assertSearchHits(searchResponse, "parent-id");
     }
+
```

Comment:
I'll add this test on top of this pr. For the 2.x branches we should still use the integ tests, since that is the only way to tests this stuff there.